nightfall-deployer/contracts/State.sol

```diff
@@ -365,7 +366,12 @@ contract State is ReentrancyGuardUpgradeable, Pausable, Key_Registry, Config {
     }
 
     function setNumProposers(uint256 np) public onlyProposer {
+        uint256 prevNumProposers = numProposers;
         numProposers = np;
+        if (numProposers == 2 && prevNumProposers == 1) {
```

Comment:
What is the significance of numProposers being `==2`?

Reply:
when changing from 1 proposer to 2 proposers is the situation where it becomes necessary to build the `spanProposersList`, to allow the `changecurrentproposer` to directly access the next proposers from the list.
We only handle the specific case that goes from 1 to 2 proposers, because all other cases are handled by the `changeCurrentProposer` function

Reply:
Got it, thanks!